dbos/core.py

```diff
@@ -150,13 +156,14 @@ def _init_workflow(
             json.dumps(ctx.authenticated_roles) if ctx.authenticated_roles else None
         ),
         "assumed_role": ctx.assumed_role,
+        "queue_name": queue,
     }
 
     # If we have a class name, the first arg is the instance and do not serialize
     if class_name is not None:
         inputs = {"args": inputs["args"][1:], "kwargs": inputs["kwargs"]}
 
-    if temp_wf_type != "transaction":
+    if temp_wf_type != "transaction" or queue is not None:
```

Comment:
Add a todo here: we should have a big transaction for update_workflow_status and update_workflow_input, and also `enqueue`. Otherwise, under some corner cases we might not be able to recover workflows, or the workflow may get stuck in "enqueued" state.